package/android/src/main/java/com/shopify/reactnative/skia/SkiaBaseView.java

```diff
@@ -138,8 +129,7 @@ private static int motionActionToType(int action) {
     @Override
     public void onSurfaceTextureAvailable(SurfaceTexture surface, int width, int height) {
         Log.i(tag, "onSurfaceTextureAvailable " + width + "/" + height);
-        mSurface = new Surface(surface);
-        surfaceAvailable(mSurface, width, height);
+        surfaceAvailable(surface, width, height);
     }
 
     @Override
```

Comment:
In `onSurfaceTextureDestroyed` it seems to be calling createSurfaceTexture - that seems a bit wrong:

```ts
@Override
    public boolean onSurfaceTextureDestroyed(SurfaceTexture surface) {
        Log.i(tag, "onSurfaceTextureDestroyed");
        // https://developer.android.com/reference/android/view/TextureView.SurfaceTextureListener#onSurfaceTextureDestroyed(android.graphics.SurfaceTexture)
        destroySurface();
        createSurfaceTexture(); // <-- CAN BE REMOVED?
        return false;
    }
```

Reply:
We will document this part which is indeed difficult to understand